### ISSUE TITLE
Infra: Improve Release Notes from Changelog?

### ISSUE THREAD

**Issue by tomschr** [python-semver/python-semver#182]
## Situation

The [Releases tab](https://github.com/python-semver/python-semver/releases) contains all of our released versions. That's fine, but it's a little bit empty. Our users could be confused what we did for this release. :wink: 

Interested users need to find the Changelog file and locate the respective version. This is a bit cumbersome.

## Proposed Solution

As we already have a Changelog, I would suggest to use this text when we release a new version. 

An example how this could look like can be found for the 2.9.0 release: https://github.com/python-semver/python-semver/releases/tag/2.9.0

Currently, it's a little bit cumbersome as the Changelog is in RST, but GitHub expects Markdown. I copied it manually and adapted the text (with the help of some nifty regex).
If you like this idea, this can be done with some scripts or already existing conversion tools to make the conversion a bit easier.

However, I don't see this as a big disadvantage as the text is usually not very long, the text can be automatically converted, and it has to be done only once.

## Benefits

I would see the following benefits:

* Interested users don't need to find the Changelog
* Everything is in one place. :wink: 
* The respective issues or pull requests are linked and can be easily followed.

**Comment by scls19fr** [comment]
Ah!!! Markdown vs reStructuredText it's like choosing between Vi and Emacs (Emacs vs VI in that case)...

I must admit that I have always prefered Emacs.

**Comment by scls19fr** [comment]
+1 about adding part of changelog for each release (especially if you have appropriate regexp to change for rst to md)

**Comment by tomschr** [comment]
Thanks! I think, this is one of the tasks that should be done for each release.

That's how I did it:

1. Open the `CHANGELOG.rst` file.
1. Copy the relevant part of the version to a separate file, for example, `v2.9.0.rst`.
1. Replace ```:gh:`(\d+)` ``` with `#\1`.
1. Replace ```:pr:`(\d+)` ``` with `PR #\1`.
1. Replace ```:func:(`[\w\._]+`)``` with `\1`.
1. Adjust the headers (I usually use two hash signs).

**Comment by scls19fr** [comment]
You can also try [pandoc](https://pandoc.org/) to convert rst to md

    pandoc CHANGELOG.rst -f rst -t markdown -o CHANGELOG.md

**Comment by tomschr** [comment]
> You can also try [pandoc](https://pandoc.org/) to convert rst to md
> 
> ```
> pandoc CHANGELOG.rst -f rst -t markdown -o CHANGELOG.md
> ```

I did that before, but I wasn't satisfied with the result. :wink: For example, I wanted to distinguish between "normal" GitHub issues (`:gh:`) and pull requests (`:pr:`). After the conversion, this information is lost (although I have to admit, it's not thaat important).

-----

Ok, I also updated the release notes for the [2.8.2 release](https://github.com/python-semver/python-semver/releases/tag/2.8.1).
